README.md

```diff
@@ -8,4 +8,249 @@ are in the process of being changed and will certainly break your code.
+        // handle error
+    }
+    ...
+
+    if err, key2 := tags.CreateKeyString("keyNameID2"); err != nil {
+        // handle error
+    }
+    ...
+
+### Create a set of tags associated with keys
+To create a new tag set from scratch using changes:
+
+    tsb := &tags.TagSetBuilder{}
+    tsb..StartFromEmpty()
+    tsb.InsertString( key1, "foo value")
+    tsb.UpdateString( key1, "foo value2")
+    tsb.UpsertString( key2, "bar value")
+    tagsSet := tsb.Build()
+
+    // A shorter way of achieving the same is:
+    tagsSet := &tags.TagSetBuilder{}.StartFromEmpty()
+                                    .InsertString( key1, "foo value")
+                                    .UpdateString( key1, "foo value2")
+                                    .UpsertString( key2, "bar value")
+                                    .Build()
+
+
+
+To create a new tagsSet from an existing tag set oldTagSet:
+    oldTagSet := ...
+    tsb := &tags.TagSetBuilder{}
+    tsb.StartFromTagSet(oldTagSet)
+    tsb.InsertString( key1, "foo value")
+    tsb.UpdateString( key1, "foo value2")
+    tsb.UpsertString( key2, "bar value")
+    newTagSet := tsb.Build()
+
+    // A shorter way of achieving the same is:
+    oldTagSet := ...
+    newTagSet := &tags.TagSetBuilder{}.StartFromTagSet(oldTagSet)
+                                      .InsertString( key1, "foo value")
+                                      .UpdateString( key1, "foo value2")
+                                      .UpsertString( key2, "bar value")
+                                      .Build()
+
+### Add new tagSet to a context / Modify tagSet in a context 
+Add tags to a context for propagation to downstream methods and downstream rpcs:
+To create a new context with the tags. This will create a new context where all the existing tags in the current context are deleted and replaced with the tags passed as argument.    
+    
+    newTagSet  := ...
+    ctx2 := tags.ContextWithNewTags(ctx, newTagSet)
+
+Create a new context keeping the old tag set and adding new tags to it, removing specific tags from it, or modifying the values fo some tags.
+    
+    oldTagSet := tags.FromContext(ctx)
+    newTagSet := &tags.TagSetBuilder{}.StartFromTagSet(oldTagSet)
+                                      .InsertString( key1, "foo value")
+                                      .UpdateString( key1, "foo value2")
+                                      .UpsertString( key2, "bar value")
+                                      .Build()
+    ctx2 := tags.ContextWithChanges(ctx, newTagSet)
+
+## Registering views and retrieving their collected data.
+
+### To create/register a measure a.k.a resource
+Create/load measures units:
+
+    // returns a *MeasureFloat64
+    mf := stats.NewMeasureFloat64("/my/float64/measureName", "some measure")
+    mi := stats.NewMeasureInt64("/my/otherName", "some other measure")
+    ...
+
+Register measures:
+
+	if err := stats.RegisterMeasure(mf); err != nil {
```

Comment:
I don't understand the difference between New and Register. Should the measures be similar to tags and just have Create methods?

Reply:
Originally I had what you are suggesting for both measures and views. But then you suggested to separate create/new from register for views in one of your comments in "Census Stats API v0.1" :)
After thinking about it, it seems separating the 2 actions makes more sense as it makes it easier to understand what is going on. Anyone can create any view/measure they want. But registering it the library is what might return an error.
I think keeping measure/views interactions consistent makes it easier to get our head around it.

Reply:
That being said I do not have a strong opinion about it. It just seems better to be consistent.

Reply:
It feels like tags and measures are the same -- kind of constants used by other things. I don't see a use case when Measure is looked up by name (same as tags). I think views are different.

Reply:
I modified measures to be automatically "registered" on creation